tests/scenarios/compat-dummy-app-test.ts

```diff
@@ -59,18 +64,54 @@ dummyAppScenarios
         await builder.build({ changedDirs: [app.dir] });
         expectFile('../../components/example.hbs').matches(/goodbye/);
       });
+    });
+
+    Qmodule(`${scenario.name} - public assets`, function (hooks) {
+      throwOnWarnings(hooks);
+
+      let app: PreparedApp;
+      let expectFile: ExpectFile;
 
-      test('contains public assets from dummy app', async function () {
-        // expectRewrittenFilesAt doesn't understand dummy apps, so even though
-        // we initialized it on app.dir/tests/dummy, we can't just say
-        // "robots.txt" here because it thinks that file belongs to the
-        // containing addon. By writing out the rewritten paths ourselves we
-        // sidestep that problem≥
+      hooks.before(async () => {
+        app = await scenario.prepare();
+      });
+
+      test('rewritten app contains public assets from both addon and dummy app after a build', async function (assert) {
+        await app.execute(`pnpm vite build`);
+        expectFile = expectRewrittenFilesAt(resolve(app.dir, 'tests/dummy'), {
+          qunit: assert,
+        });
+        expectFile('../../node_modules/.embroider/rewritten-app/addon-template/from-addon.txt').exists();
         expectFile('../../node_modules/.embroider/rewritten-app/robots.txt').exists();
-        expectFile('../../node_modules/.embroider/rewritten-app/package.json')
+        let assets = expectFile('../../node_modules/.embroider/rewritten-app/package.json')
           .json()
-          .get('ember-addon.assets')
-          .includes('robots.txt');
+          .get('ember-addon.assets');
+        assets.includes('robots.txt');
+        assets.includes('./addon-template/from-addon.txt');
+      });
+
+      test('production build contains public assets from both addon and dummy app after a build', async function (assert) {
```

Comment:
This test builds the dummy app the same way as the one checking the rewritten-app so I could have done all the checks in the same test. I preferred separated tests because I want to remove the rewritten-app one in PR #1859